### DISCUSSION TITLE
Stylesheets/CSS not loading

### DISCUSSION THREAD

**Discussion by ju5t** [spatie/browsershot#584] [Q&A]
In our application we can send PDF's via e-mail. Today, I noticed that emails sent through my tinker session had no stylesheets applied to them. When I sent the exact same email from a browser session, it's OK.

So here's what I did to test it further:
- I created a new command; just to be sure it's not something related to tinker. It didn't show the stylesheets.
- I then create a simple function based `Route::get`, pasting in the same code. This did show the stylesheets.

Apparently something is different when I run a command/tinker vs. a browser session. I couldn't find anything in the docs or previous discussions. I am using the latest version.

Am I doing something wrong? Or can someone point me in the right direction to debug this further?

**Comment by chrisbalicki** [comment]
I am having a similar issue and have not got to the bottom of it yet however when adding the CSS as a CDN it seems to work fine so it seems like its something to do with loading the css file...

**Reply by ju5t** [reply]
In #516 there is a discussion about timeouts on local uri's. This is not the same issue (at least not for me, as I haven't seen those timeouts), but the solution in that discussion did fix it for me:

```blade
@env('local')
      @php
            echo '<style>' . file_get_contents(realpath(public_path('css/app.css'))) . '</style>';
      @endphp
@endenv
@env(['staging', 'production'])
    <link href="{{ asset('css/app.css') }}" rel="stylesheet">
@endenv
```

credit to @akhan619.

It's not ideal, but if you need a quick fix, this is it.

**Reply by thursdaydan** [reply]
I'm having a similar problem.

For a while my PDF exports were working with styles on local and remote servers, then they stopped including styles and images on the remote servers for some reason.

Local PDFs still apply all of the styles and images as expected though, so not sure if it's a server config issue (at least in my case.)

**Reply by davidwhicker** [reply]
Did you ever figure this one out? Same issue here, local is fine but remote is not ( forge server )
